plugin/nerdcommenter.vim

```diff
@@ -125,7 +125,7 @@ function! NERDCommentIsCharCommented(line, col)
     return nerdcommenter#IsCharCommented(a:line, a:col)
 endfunction
 
-inoremap <silent> <Plug>NERDCommenterInsert <Space><BS><Esc>:call nerdcommenter#Comment('i', "insert")<CR>
+inoremap <silent> <Plug>NERDCommenterInsert <C-\><C-O>:call nerdcommenter#Comment('i', "Insert")<CR>
```

Comment:
Can you explain the change in key sequences here? I'm concerned this might have side effects like being affected by user `<Leader>` key settings.

Reply:
`<C-\><C-O>` (`:h i_CTRL-\_CTRL-O`) enters the "Insert Normal mode" that lets you run a single normal mode command and return to insert mode. In this state, `:` (which is a normal mode command) enters the cmdline mode to run `call ...`. After `call ...` command finishes, vim will return to insert mode. 

The old version exits insert mode (`<Esc>`), and returns to insert mode using `startinsert!` or `feedkeys('a', 'ni')` (in `s:PlaceDelimitersAndInsBetween`). I'm not sure what `<Space><BS>` is for. It predates the history of this repository..

Reply:
Thanks for the explanation. I am guessing `<Space><BS>` was intended as a hack to have an undo point, but whether that guess is correct or not I suspect it was and is the wrong solution to whatever problem it was meant to address.